app/actions/configurations.py

```diff
@@ -0,0 +1,37 @@
+from datetime import datetime
+from enum import Enum
+from typing import List, Optional
+
+from pydantic import Field
+
+from .core import AuthActionConfiguration, PullActionConfiguration, ExecutableActionMixin
+
+
+class AuthenticateConfig(AuthActionConfiguration, ExecutableActionMixin):
+    username: Optional[str] = Field(
+        "",
+        example="user@pamdas.org",
+        description="Username used to authenticate against Earth Ranger API",
+    )
+    password: Optional[str] = Field(
+        "",
+        example="passwd1234abc",
+        description="Password used to authenticate against Earth Ranger API",
+    )
+    token: Optional[str] = Field(
+        "",
+        example="1b4c1e9c-5ee0-44db-c7f1-177ede2f854a",
+        description="Token used to authenticate against Earth Ranger API",
+    )
```

Comment:
I prerfer SecretStr for these, just in case the object gets written to logs for example.

Reply:
I absolutely agree, thanks for catching that. I reused code from a POC I wrote long ago, before knowing about `SecretStr` and I missed updating that.